stories/form/form-masked-input/form-masked-input.stories.mdx

```diff
@@ -0,0 +1,173 @@
+import { Canvas, Meta, Story } from "@storybook/addon-docs";
+import { useState } from "react";
+import { MaskedInput } from "src/masked-input";
+import { Form } from "src/form";
+import { Text } from "src/text";
+import {
+    Heading3,
+    Heading4,
+    Secondary,
+    StoryContainer,
+    Title,
+} from "../../storybook-common";
+import { Container } from "../shared-doc-elements";
+import { PropsTable } from "./props-table";
+import { ToggleOffFillIcon } from "@lifesg/react-icons/toggle-off-fill";
+import { ToggleOffIcon } from "@lifesg/react-icons/toggle-off";
+
+<Meta title="Form/MaskedInput" component={Form.MaskedInput} />
+
+<Title>Form.MaskedInput</Title>
+
+<Secondary>Overview</Secondary>
+
+An enhanced field that allows a user to enter a response that is masked on blur.
+
+```tsx
+import { Form } from "@lifesg/react-design-system/form";
+```
+
+<Canvas>
+    <Story name="Field">
+        <StoryContainer>
+            <Container>
+                <Form.MaskedInput
+                    label="This is the mask field with no input"
+                    maskRange={[2, 5]}
+                />
+                <Form.MaskedInput
+                    label="This is the mask field with input"
+                    value="S1234567D"
+                    maskRange={[2, 5]}
+                />
+                <Form.MaskedInput
+                    label="This is the error state"
+                    errorMessage="Input is required"
+                    maskRange={[2, 5]}
+                />
```

Comment:
there was this AC mentioned:
> If the input fails validation, the input should NOT be masked. The mask icon will remain disabled. Masking only occurs when the input is correct and meets the validation requirements.

just to confirm, it's up to the consumer to conditionally set the mask config depending on the value? (this seems to be the most flexible way, instead of hardcoding it to `error === true`)

Reply:
ya that makes sense. in that case should we also allow consumers to configure the mask icon disabled state as well or should we toggle it depending on whether the value is masked or not?

Reply:
Hmm yeah it should be handled consistently. If masking is disabled, the icon should also be disabled. How about a `disableMask` boolean (default false) that disables masking entirely?